[FIX] Recommend SI units formatting to adhere to CMIXF-12 (#411)
bids-standard/bids-specification: src/02-common-principles.md

description of the corresponding column, using an object containing the following
fields:

| Field name  | Definition                                                                                                                        |
| :---------- | :-------------------------------------------------------------------------------------------------------------------------------- |
| LongName    | Long (unabbreviated) name of the column.                                                                                          |
| Description | Description of the column.                                                                                                        |
| Levels      | For categorical variables: a dictionary of possible values (keys) and their descriptions (values).                                |

> | Units       | Measurement units. SI units in CMIXF formatting are REQUIRED when possible (see [units section](./02-common-principles.md#units). |

Author: could it be that you performed parts of your review on an outdated commit @effigies?

the sidecar JSON file. In case non-standard prefixes are added to SI or non-SI
units, these non-standard prefixed units MUST be specified in the JSON file.
See [Appendix V](99-appendices/05-units.md) for a list of standard units and
prefixes. Note also that for the *formatting* of SI units, the
[CMIXF-12](https://people.csail.mit.edu/jaffer/MIXF/CMIXF-12) convention for

> encoding units is RECOMMENDED.

Author: ```suggestion
encoding units is REQUIRED.
```

Reviewer: can we add an example here? What would be CMIXF-12 compatible and what would not be? Just to make sure that users don't get lost in trying to understand another standard ...

Author: Good point, ... there are some examples in the appendix, but I'll also add some here. :+1:

## Units

All units SHOULD be specified as per [International System of Units](https://en.wikipedia.org/wiki/International_System_of_Units)
(abbreviated as SI, from the French Système international (d'unités)) and can
be SI units or [SI derived units](https://en.wikipedia.org/wiki/SI_derived_unit).
In case there are valid reasons to deviate from SI units or SI derived units,
the units MUST be specified in the sidecar JSON file.
In case data is expressed in SI units or SI derived units, the units MAY be
specified in the sidecar JSON file.
In case non-standard prefixes are added to SI or non-SI units, these
non-standard prefixed units MUST be specified in the JSON file.
See [Appendix V](99-appendices/05-units.md) for a list of standard units and
prefixes.
Note also that for the *formatting* of SI units, the [CMIXF-12](https://people.csail.mit.edu/jaffer/MIXF/CMIXF-12)
convention for encoding units is REQUIRED.
More information about CMIXF can be found in [Appendix V](99-appendices/05-units.md)
but to provide some examples, CMIXF formatting of "micro volts" would be `uV`,

> and of "kilobecquerel per ml" would be `kBq/ml`.

Reviewer 2: ```suggestion
and of "kilobecquerel per milliliter" would be `kBq/mL`.
```

> and of "kilobecquerel per milliliter" would be `kBq/mL`.

Reviewer: I think we should also give an example of what would be an invalid CMIXF formatting.

Author: There is:

 1.  `uV` or `µV` are permissible, but NOT: `microV`, `µvolt`, `1e-6V`, etc.

a bit further below, I think that suffices, but feel free to make me do it if you feel strongly about it

Reviewer: yeah I noticed once I read the whole thing. I think CMIXF comes out of the blue a bit. If there was an example to motivate why to use CMIXF it may help. But it's up to you.

> and of "kilobecquerel per milliliter" would be `kBq/mL`.

Reviewer 3: Well, if we're not in a rush to approve, can I suggest smoother wording here? "... but to provide..." is weird. Better would be to sketch out what CMIXF is trying to do, then provide an example; also, "kBq/mL" isn't so informative b/c that's a routine abbreviation -- better to focus on the exceptions that might be counter intuitive.

What about this text for this passage, starting after "convention for encoding units is REQUIRED."

> CMIXF provides a consistent system for all units and prefix symbols with only basic characters, avoiding symbols that can cause text encoding problems; for example the CMIXF formatting for "micro volts" is `uV`, "degrees Celsius" is `oC` and "Ohm" is `Ohm`. See [Appendix V](99-appendices/05-units.md) for more information.

Happy to make a PR if people like this.

Author: > if we're not in a rush to approve

definitely not in a rush, I am just trying to keep the momentum up so that this PR does not go stale for another year. If we see problems or ways to make it better, we should (and will) take the time to make it as good as we can given all constraints.

> Happy to make a PR if people like this.

PRs are welcome to: https://github.com/sappelhoff/bids-specification/tree/asciiunits

an alternative for PRs is to use the [GitHub suggestion function](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/reviewing-proposed-changes-in-a-pull-request) --> multiline suggestions are now also possible [gh-blog-post on this](https://github.blog/changelog/2020-02-26-multi-line-code-suggestions-beta/)

Reviewer 3: Cool!  [Done](https://github.com/bids-standard/bids-specification/pull/411#discussion_r432484588).

Author: Thanks, this is a lot better.

> Note also that for the *formatting* of SI units, the [CMIXF-12](https://people.csail.mit.edu/jaffer/MIXF/CMIXF-12)
> convention for encoding units is REQUIRED.
> More information about CMIXF can be found in [Appendix V](99-appendices/05-units.md)
> but to provide some examples, CMIXF formatting of "micro volts" would be `uV`,
> and of "kilobecquerel per milliliter" would be `kBq/mL`.

Reviewer 3: ```suggestion
CMIXF provides a consistent system for all units and prefix symbols with only basic characters, avoiding symbols that can cause text encoding problems; for example the CMIXF formatting for "micro volts" is uV, "degrees Celsius" is oC and "Ohm" is Ohm. See Appendix V for more information.
```

> More information about CMIXF can be found in [Appendix V](99-appendices/05-units.md)
> but to provide some examples, CMIXF formatting of "micro volts" would be `uV`,
> and of "kilobecquerel per milliliter" would be `kBq/mL`.

Reviewer 3: ```suggestion
CMIXF provides a consistent system for all units and prefix symbols with only basic
characters, avoiding symbols that can cause text encoding problems; for example the
CMIXF formatting for "micro volts" is uV, "degrees Celsius" is oC and "Ohm" is Ohm. 
See Appendix V for more information.
```

description of the corresponding column, using an object containing the following
fields:

> | Field name  | Definition                                                                                                             |
> | :---------- | :--------------------------------------------------------------------------------------------------------------------- |
> | LongName    | Long (unabbreviated) name of the column.                                                                               |
> | Description | Description of the column.                                                                                             |
> | Levels      | For categorical variables: a dictionary of possible values (keys) and their descriptions (values).                     |
> | Units       | Measurement units. SI units in CMIXF formatting are RECOMMENDED (see [units section](./02-common-principles.md#units). |
> | TermURL     | URL pointing to a formal definition of this type of data in an ontology available on the web.                          |

Reviewer 2: ```suggestion
| Field name  | Definition                                                                                                      |
| :---------- | :-------------------------------------------------------------------------------------------------------------- |
| LongName    | Long (unabbreviated) name of the column.                                                                        |
| Description | Description of the column.                                                                                      |
| Levels      | For categorical variables: a dictionary of possible values (keys) and their descriptions (values).              |
| Units       | Measurement units. SI units in CMIXF formatting are RECOMMENDED (see [Units](./02-common-principles.md#units)). |
| TermURL     | URL pointing to a formal definition of this type of data in an ontology available on the web.                   |
```

Describing dates and timestamps:

-   Date time information MUST be expressed in the following format
    `YYYY-MM-DDThh:mm:ss[.000000]` (year, month, day, hour (24h), minute,
    second, and optionally fractional second).
    This is equivalent to the RFC3339 "date-time" format, time zone is always
    assumed as local time).
    No specific precision is required for fractional seconds, but the precision
    SHOULD be consistent across the dataset.
    For example `2009-06-15T13:45:30`

-   Time stamp information MUST be expressed in the following format:
    `13:45:30[.000000]`

-   Dates can be shifted by a random number of days for privacy protection
    reasons.

>     To distinguish real dates from shifted dates always use year 1925

Reviewer: ```suggestion
    To distinguish real dates from shifted dates, always use year 1925
```
